dotCMS/src/integration-test/java/com/dotcms/security/apps/AppsAPIImplTest.java

```diff
@@ -1035,43 +1074,180 @@ public void Test_File_Comparison_Is_Case_Sensitive()
      */
     @Test
     public void Test_Create_Secrets_Then_Export_Them_Then_Import_Then_Save()
-            throws DotDataException, DotSecurityException, IOException, EncryptorException, ClassNotFoundException {
+            throws DotDataException, DotSecurityException, IOException, EncryptorException, AlreadyExistException {
         final User admin = TestUserUtils.getAdminUser();
         final Host site = new SiteDataGen().nextPersisted();
         final AppsAPI api = APILocator.getAppsAPI();
+        //generate a descriptor
         final AppSecrets.Builder builder1 = new AppSecrets.Builder();
-        final String appKey = "appKey-1-Host-1";
-        final AppSecrets secrets = builder1.withKey(appKey)
-                .withHiddenSecret("test:secret1", "secret-1")
-                .withHiddenSecret("test:secret2", "secret-2")
-                .withHiddenSecret("test:secret3", "secret3")
-                .withHiddenSecret("test:secret4", "secret-4")
+        final AppDescriptorDataGen dataGen = new AppDescriptorDataGen()
+                .stringParam("p1", false,  true)
+                .stringParam("p2", false,  true)
+                .withName("system-app-example")
+                .withDescription("system-app")
+                .withExtraParameters(false);
+        final File file = dataGen.nextPersistedDescriptor();
+        api.createAppDescriptor(file, admin);
+
+        final String appKey = dataGen.getKey();
+        //generate secrets
+        AppSecrets secrets = builder1.withKey(appKey)
+                .withHiddenSecret("p1", "secret-1")
+                .withHiddenSecret("p2", "secret-2")
                 .build();
         //Save it
         api.saveSecrets(secrets, site, admin);
         final Optional<AppSecrets> secretsOptional = api.getSecrets(appKey, site, admin);
-        final AppSecrets appSecretsPostSave = secretsOptional.get();
+        Assert.assertTrue(secretsOptional.isPresent());
+        secrets = secretsOptional.get();
 
-        //AES only supports key sizes of 16, 24 or 32 bytes.
+        //AES only supports security Key of sizes of 16, 24 or 32 bytes.
         final String password = RandomStringUtils.randomAlphanumeric(32);
-        final Key key = AppsUtil.generateKey(password);
-        final Path exportSecretsFile = api.exportSecrets(key, true, null, admin);
+        final Key securityKey = AppsUtil.generateKey(password);
+
+        //Now that we have a valid key lets dump our selection of secrets
+        final Map<String,Set<String>> appKeysBySite = ImmutableMap.of(site.getIdentifier(), ImmutableSet.of(appKey));
```

Comment:
Codacy found an issue: [Unnecessary use of fully qualified name 'ImmutableMap.of' due to existing static import 'com.google.common.collect.ImmutableMap.of'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6286472)